gno.land/pkg/gnoclient/signer.go

```diff
@@ -0,0 +1,144 @@
+package gnoclient
+
+import (
+	"fmt"
+
+	"github.com/gnolang/gno/gno.land/pkg/sdk/vm"
+	"github.com/gnolang/gno/tm2/pkg/crypto/keys"
+	"github.com/gnolang/gno/tm2/pkg/errors"
+	"github.com/gnolang/gno/tm2/pkg/std"
+)
+
+// Signer provides an interface for signing transactions.
+type Signer interface {
+	Sign(SignCfg) (*std.Tx, error) // Signs a transaction and returns a signed tx ready for broadcasting.
+	Info() keys.Info               // Returns key information, including the address.
+	Validate() error               // Checks whether the signer is properly configured.
+}
+
+// SignerFromKeybase represents a signer created from a Keybase.
+type SignerFromKeybase struct {
+	Keybase  keys.Keybase // Stores keys in memory or on disk
```

Comment:
Just a note, for implementing tx signing functionality, we don't need the bloated keybase necessarily. Initializing it and signing (even in-memory) has a performance cost that is not negligible. 
We don't need to rework the signing mechanism in this PR, just to keep in mind there are alternatives:
https://github.com/gnolang/faucet/blob/main/sign.go

Reply:
Does it make sense to keep both implementation ?

Reply:
We use SignerFromKeybase in Gno Native Kit because of its on-disk storage. https://github.com/gnolang/gnonative/blob/d07ccc50841dacf01fda21b45fc17c2f279ea5e2/service/service.go#L108
and assign it as the Signer in gnoclient.Client
https://github.com/gnolang/gnonative/blob/d07ccc50841dacf01fda21b45fc17c2f279ea5e2/service/service.go#L117